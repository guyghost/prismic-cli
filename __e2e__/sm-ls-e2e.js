const path = require('path');
const fs = require('fs');
const { spawnSync } = require('child_process');
const {
  setup,
  TMP_DIR,
  PRISMIC_BIN,
  rmdir,
  mkdir,
  genRepoName,
  RETRY_TIMES,
} = require('./utils');


describe('prismic sm --ls', () => {
  jest.retryTimes(RETRY_TIMES);
  jest.setTimeout(300000);

  const repoName = genRepoName('cli-sm-ls-test');
  const dir = path.resolve(TMP_DIR, 'sm-ls');

  beforeAll(async () => {
    return rmdir(dir, { recursive: true })
    .then(() => mkdir(TMP_DIR, { recursive: true }))
    .then(() => setup(repoName));
  });

  it('should list the slices available in a project', () => {

    const themeArgs = [
      'theme',
      '--theme-url', 'https://github.com/prismicio/nuxtjs-blog.git',
      '--conf', 'nuxt.config.js',
      '--domain', repoName,
      '--folder', dir,
      '--skip-install',
    ];

    spawnSync(PRISMIC_BIN, themeArgs, { encoding: 'utf8' });
    expect(fs.existsSync(dir)).toBe(true);

    const setupArgs = ['sm', '--setup', '--domain', repoName, '--yes'];
    spawnSync(`pushd ${dir} && ${PRISMIC_BIN}`, setupArgs, { encoding: 'utf8', shell: true });
    const smfile = path.resolve(dir, 'sm.json');
    expect(fs.existsSync(smfile)).toBe(true);

    const args = ['sm', '--ls'];
    const cmd = `pushd ${dir} && ${PRISMIC_BIN}`;

    const res = spawnSync(cmd, args, { encoding: 'utf8', shell: true });
    expect(res.stdout).toBeTruthy();
    expect(res.stderr).toBeFalsy();
    expect(res.status).toBeFalsy();
  });
});
