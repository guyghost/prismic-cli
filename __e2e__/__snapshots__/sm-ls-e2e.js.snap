// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prismic sm --ls should list the slices available in a project 1`] = `
"~/Projects/prismic/prismic-cli/__tmp__/sm-ls ~/Projects/prismic/prismic-cli
[info] Found 10 slices across 2 libraries:

ðŸ¥’ at vue-essential-slices:
  * AlternateGrid
  * CallToAction
  * CardsCarousel
  * CustomerLogos
  * FaqSection
  * FeatureTestimonials
  * ImagesSlider
  * PricingTable
  * TestimonialsSlider
  * VideoHighlights

[info] Something's missing?
Check paths in your configuration file!

"
`;

exports[`prismic sm --ls should list the slices available in a project 2`] = `""`;
